### PR TITLE
[PIPE-314] Create summary_state_view as Spark ETL

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**Description:**
Implemented the `summary_state_view` matview SQL as a Spark query.

**Technical details:**
Copied over and made minor changes to support the current `summary_state_view` matview SQL as Spark SQL.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [PIPE-314](https://federal-spending-transparency.atlassian.net/browse/PIPE-314):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

MERGE AFTER #3517 

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Review by kwhickey — CHANGES_REQUESTED** [review]
A few questions before final approval.

File: usaspending_api/transactions/delta_models/summary_state_view.py [hunk 1]
```diff
@@ -20,3 +20,91 @@
...
+                0
+            ) AS NUMERIC(23, 2)
+        ) AS face_value_loan_guarantee,
+        COUNT(*) AS counts
+    FROM
+        raw.transaction_normalized
+    LEFT OUTER JOIN
+        raw.transaction_fpds ON (transaction_normalized.id = transaction_fpds.transaction_id)
+    LEFT OUTER JOIN
+         raw.transaction_fabs ON (transaction_normalized.id = transaction_fabs.transaction_id)
```
[the remark this answers is not in the document]
Comment (sethstoudenmier) [reply usaspending_api/transactions/delta_models/summary_state_view.py:92]: Will make this change once I go to update the PR following merge of other PRs.

File: usaspending_api/transactions/delta_models/summary_state_view.py [hunk 2]
```diff
@@ -20,3 +20,91 @@
...
+        REGEXP_REPLACE(
+            MD5(
+                CONCAT_WS(
+                    ' ',
+                    SORT_ARRAY(COLLECT_LIST(transaction_normalized.id))
+                )
+            ),
+            '^(\.{{{{8}}}})(\.{{{{4}}}})(\.{{{{4}}}})(\.{{{{4}}}})(\.{{{{12}}}})$',
+            '\$1-\$2-\$3-\$4-\$5'
+        ) AS duh,
```
[the remark this answers is not in the document]
Comment (sethstoudenmier) [reply usaspending_api/transactions/delta_models/summary_state_view.py:39]: It is the current way used to create the `duh` value. Not the biggest fan of it, but it is consistent with current implementation.

[the remark this answers is not in the document]
Comment (kwhickey) [reply usaspending_api/transactions/delta_models/summary_state_view.py:39]: Ok. Can you put a TODO comment inline the sql suggesting a better way to do this key as a future improvement. if e.g. you think the agg_key approach would work.

[the remark this answers is not in the document]
Comment (sethstoudenmier) [reply usaspending_api/transactions/delta_models/summary_state_view.py:40]: Added a comment

**Review by kwhickey — APPROVED** [review]
Good and clean! Approving.

**Pull request merged**